renovate.json

```
@@ -22,6 +22,17 @@
       "depNameTemplate": "renovate/renovate",
       "datasourceTemplate": "docker"
     },
+    {
+      "fileMatch": ["(^|/)README.md$"],
+      "matchStrings": [
+        "Version: (?<currentValue>.*?)",
+        "Version-(?<currentValue>.*?)",
+        "AppVersion-(?<currentValue>.*?)",
+        "AppVersion: (?<currentValue>.*?)"
```

```suggestion
        "(\\s|-)(?<currentValue>\\d+\\.\\d+\\.\\d+)"
```
I think this is a simpler fix (only one regex)

---

I merged the suggestion.

But the `\\d+\\.\\d+\\.\\d+`-part is to catch what exactly?

---

Simply match a version like `23.85.1` .

So match `\d+` matches one or more digets. `\.` matches a dot, the double backslash is for string escape.

🙃

---

So much I know, but I thought that `<currentValue>` were the same regex.  And so I could not understand why you needed to provide the regex twice.

Do they complement each other, or do you need to pass the regex explicitly?  I'm not quite sure how `<currentValue>` works.

Thanks for the explanation though. 😄

---

`(?<currentValue>)` is a javascript regex named capture group. `.*?` mean match anything, but less as possible. So my regex is now more explicit serching for a version like string, so we don't need to much prefix  / suffix matching.

---

Oh.  Would you look at that!  The more you learn. 😉 

Thanks again.

---

Yeah, https://regex101.com/ is a good help for testing regexes 🙃